### PR TITLE
[User Model] Bind to iOS SDK

### PR DESCRIPTION
**Pull request by shepherd-l**
# Description
## One Line Summary
Bind Unity SDK to iOS SDK

## Details

# Affected code checklist
   - [ ] Notifications
      - [ ] Display
      - [ ] Open
      - [ ] Push Processing
      - [ ] Confirm Deliveries
   - [ ] Outcomes
   - [ ] Sessions
   - [ ] In-App Messaging
   - [ ] REST API requests
   - [x] Public API changes

# Checklist
## Overview
   - [x] I have filled out all **REQUIRED** sections above
   - [x] PR does one thing
   - [x] Any Public API changes are explained in the PR details and conform to existing APIs

## Testing
   - [x] I have included test coverage for these changes, or explained why they are not needed
   - [x] All automated tests pass, or I explained why that is not possible
   - [x] I have personally tested this on my device, or explained why that is not possible

## Final pass
   - [x] Code is as readable as possible.
   - [x] I have reviewed this PR myself, ensuring it meets each checklist item

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Unity-SDK/582)

### CODE REVIEW

File: com.onesignal.unity.ios/Runtime/iOSNotificationsManager.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        [AOT.MonoPInvokeCallback(typeof(NotificationWillShowInForegroundDelegate))]
+        private static bool _onWillShow(string response) {
+            if (_instance.WillShow == null)
+                return true;
+
+            var notification = JsonUtility.FromJson<Notification>(response);
+            _fillNotifFromObj(ref notification, Json.Deserialize(response));
+
+            Notification resultNotif = null;
+            UnityMainThreadDispatch.Send(state => { resultNotif = _instance.WillShow(notification);});
```
Comment (brismithers) [inline com.onesignal.unity.ios/Runtime/iOSNotificationsManager.cs:101]: can you add a comment here that explains why we need to block the current thread using `UnityMainThreadDispatch.Send`.

**Review by brismithers — APPROVED** [review]
All my comments can be addressed in a future PR.  If possible @nan-li could you take a deeper look at the iOS bridges and make sure it is consistent?  Thanks!

File: com.onesignal.unity.ios/Runtime/Plugins/iOS/OneSignalBridgeUtil.mm [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/*
```
[the remark this answers is not in the document]
Comment (shepherd-l) [reply com.onesignal.unity.ios/Runtime/Plugins/iOS/OneSignalBridgeUtil.mm:1]: It's used for the `jsonStringFromDictionary` method

File: com.onesignal.unity.ios/Runtime/OneSignaliOS.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+            if (_session == null) {
+                _session = new iOSSessionManager();
+            }
+
+            _completedInit(appId);
+        }
+
+        public override void Login(string externalId, string jwtBearerToken = null) {
+            _login(externalId);
```
[the remark this answers is not in the document]
Comment (shepherd-l) [reply com.onesignal.unity.ios/Runtime/OneSignaliOS.cs:150]: Adding in another PR

File: com.onesignal.unity.ios/Runtime/iOSNotificationsManager.cs (hunk 1 above)
Comment (shepherd-l) [reply com.onesignal.unity.ios/Runtime/iOSNotificationsManager.cs:101]: Adding in another PR

**Pull request merged**